lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -724,6 +725,91 @@ def cmd_services_help(default_columns)
     [ '-h', '--help' ] => [ false, 'Show this help information.' ]
   )
 
+  def db_connection_info(framework)
+    unless framework.db.connection_established?
+      return "#{framework.db.driver} selected, no connection"
+    end
+
+    cdb = ''
+    if framework.db.driver == 'http'
+      cdb = framework.db.name
+    else
+      ::ApplicationRecord.connection_pool.with_connection do |conn|
+        if conn.respond_to?(:current_database)
+          cdb = conn.current_database
+        end
+      end
+    end
+
+    if cdb.empty?
+      output = "Connected Database Name could not be extracted. DB Connection type: #{framework.db.driver}."
+    else
+      output = "Connected to #{cdb}. Connection type: #{framework.db.driver}."
+    end
+
+    output
+  end
+
+  def cmd_stats(*args)
+    return unless active?
+    print_line "Session Type: #{db_connection_info(framework)}"
+
+    tbl = Rex::Text::Table.new(
+        'Header'     => 'Database Stats',
+        'Columns'    => ['ID', 'Hosts', 'Vulnerabilities', 'Notes', 'Services'],
+    )
+
+    current_workspace = framework.db.workspace
+    example_workspaces = ::Mdm::Workspace.order(id: :desc).take(10)
+    ordered_workspaces = ([current_workspace] + example_workspaces).uniq.sort_by(&:id)
+    tbl = Rex::Text::Table.new(
+    'Indent'  => 2,
+    'Header'  => "Database Stats",
+    'Columns' =>
+      [
+        "Active",
+        "ID",
+        "Name",
+        "Hosts",
+        "Services",
+        "Services per Host",
+        "Vulnerabilities",
+        "Vulnerabilities per Host",
+        "Notes",
+      ],
+    )
+    tbl.sort_index = 1 # ID
+    ordered_workspaces.map do |workspace|
+      active = current_workspace.id == workspace.id ? "=>" : ''
```

Comment:
I think you use stylers this will remove some of the excess spacing that's present in your screenshot

https://github.com/rapid7/metasploit-framework/pull/16946

Reply:
nice!